index.js

```diff
@@ -195,11 +210,14 @@ app.post('/users',
   (required)
   BirthDate: Date
 }*/
-app.put('/users/:Username', passport.authenticate('jwt', { session: false }), (req, res) => {
-    if (req.user.Username !== req.params.Username) {
-        return res.status(400).send('Permission denied');
-    }
-    Users.findOneAndUpdate({ Username: req.params.Username },
+app.put('/users', passport.authenticate('jwt', { session: false }), (req, res) => {
+  //dont do this
+    // if (req.user.Username !== req.params.Username) {
```

Comment:
You should just use the user form the token for these type of requests. Dont allow the user tell you who they are. Only the token should tell you that